development/tools/cmd/githubstats/main.go

```diff
@@ -0,0 +1,183 @@
+
+	variables["state"] = githubv4.IssueState("CLOSED")
+	err = client.Query(context.Background(), &query_no_labels, variables)
+	if err != nil {
+		log.Fatal(err)
+	}
+	r.Issues.Closed.TotalCount = query_no_labels.Repository.Issues.TotalCount
+
+	// Closed
+	variables["labels"] = []githubv4.String{"bug"}
+	r.Issues.Closed.Bugs = executeQuery(client, variables)
+
+	variables["labels"] = []githubv4.String{"regression"}
+	r.Issues.Closed.Regressions = executeQuery(client, variables)
+
+	variables["labels"] = []githubv4.String{"test-failing"}
+	r.Issues.Closed.TestFailing = executeQuery(client, variables)
+
+	variables["labels"] = []githubv4.String{"test-missing"}
+	r.Issues.Closed.TestMissing = executeQuery(client, variables)
+
+	variables["labels"] = []githubv4.String{"priority/critical"}
+	r.Issues.Closed.PriorityCritical = executeQuery(client, variables)
+
+	// Open
+	variables["state"] = githubv4.IssueState("OPEN")
+	variables["labels"] = []githubv4.String{"bug"}
+	r.Issues.Open.Bugs = executeQuery(client, variables)
+
+	variables["labels"] = []githubv4.String{"regression"}
+	r.Issues.Open.Regressions = executeQuery(client, variables)
+
+	variables["labels"] = []githubv4.String{"test-failing"}
+	r.Issues.Open.TestFailing = executeQuery(client, variables)
+
+	variables["labels"] = []githubv4.String{"test-missing"}
+	r.Issues.Open.TestMissing = executeQuery(client, variables)
+
+	variables["labels"] = []githubv4.String{"priority/critical"}
+	r.Issues.Open.PriorityCritical = executeQuery(client, variables)
+
+	json, err := json.Marshal(r)
+	fatalOnError(err, "while marshaling json")
+	fmt.Println(string(json))
+}
+
+func executeQuery(client *githubv4.Client, variables map[string]interface{}) int64 {
+	err := client.Query(context.Background(), &query, variables)
+	if err != nil {
+		fatalOnError(err, "while executing query")
+	}
+	return query.Repository.Issues.TotalCount
+}
+
+func fatalOnError(err error, context string) {
+	if err != nil {
+		log.Fatal(errors.Wrap(err, context))
+	}
+}
+
+func main() {
+	cfg := Config{}
+	var rootCmd = &cobra.Command{
+		Use:   "githubstats",
+		Short: "githubstats fetches stats for github issues",
+		Long:  `githubstats fetches stats for github issues and prints JSON report`,
+		Run: func(cmd *cobra.Command, args []string) {
+			getStats(cfg)
+		},
+	}
+
+	rootCmd.PersistentFlags().StringVarP(&cfg.GithubAccessToken, "github-access-token", "t", "", "github token [Required]")
+	rootCmd.PersistentFlags().StringVarP(&cfg.GithubRepoOwner, "github-repo-owner", "o", "", "owner/organisation name [Required]")
+	rootCmd.PersistentFlags().StringVarP(&cfg.GithubRepoName, "github-repo-name", "r", "", "repository name [Required]")
+
+	rootCmd.MarkPersistentFlagRequired("github-token")
+	rootCmd.MarkPersistentFlagRequired("owner")
+	rootCmd.MarkPersistentFlagRequired("repo")
+
+	envy.ParseCobra(rootCmd, envy.CobraConfig{Prefix: "APP", Persistent: true, Recursive: false})
+
+	if err := rootCmd.Execute(); err != nil {
```

Comment:
use just `fatalOnError`

Reply:
I think in this case I could use just os.Exit(), from what I can see Execute() already prints error to stderr
With os.Exit:
```
Error: required flag(s) "github-repo-name", "github-repo-owner" not set
Usage:
  githubstats [flags]

Flags:
  -t, --github-access-token string   github token [Required] [APP_GITHUB_ACCESS_TOKEN]
  -r, --github-repo-name string      repository name [Required] [APP_GITHUB_REPO_NAME]
  -o, --github-repo-owner string     owner/organisation name [Required] [APP_GITHUB_REPO_OWNER]
  -h, --help                         help for githubstats

exit status 1
```
with fatalOnError:
```
Error: required flag(s) "github-repo-name", "github-repo-owner" not set
Usage:
  githubstats [flags]

Flags:
  -t, --github-access-token string   github token [Required] [APP_GITHUB_ACCESS_TOKEN]
  -r, --github-repo-name string      repository name [Required] [APP_GITHUB_REPO_NAME]
  -o, --github-repo-owner string     owner/organisation name [Required] [APP_GITHUB_REPO_OWNER]
  -h, --help                         help for githubstats

2019/12/09 19:37:02 : required flag(s) "github-repo-name", "github-repo-owner" not set
```
the error message is repeated